File: Python/231.Power of Two.py
#判断一个数是否是2的次方
class Solution(object):
    def isPowerOfTwo(self, n):
        if n==0:    return False
        while n & 1==0:
            n = n>>1
        return n==1

s = Solution()
print(s.isPowerOfTwo(0))

File: C++/Math/231.Power of Two.cpp
/*
判断一个数是不是2的乘方。
*/
/*
思路：return !(n&(n-1));
n和n-1的按位与是否得0。
 */
#include <iostream>
#include <cstring>
using namespace std;

class Solution {
public:
	bool isPowerOfTwo(int n) {
        while(!(n & 1))
        	n = n >> 1;
        return (n>>1) == 0;
    }
};
int main()
{
	Solution s;
	cout<<s.isPowerOfTwo(130)<<endl;
}

